eth/catalyst/simulated_beacon.go

```diff
@@ -261,6 +273,50 @@ func (c *SimulatedBeacon) setCurrentState(headHash, finalizedHash common.Hash) {
 	}
 }
 
+// Commit seals a block on demand.
+func (c *SimulatedBeacon) Commit() common.Hash {
+	withdrawals := c.withdrawals.gatherPending(10)
+	if err := c.sealBlock(withdrawals, uint64(time.Now().Unix())); err != nil {
+		log.Warn("Error performing sealing work", "err", err)
+	}
+	return c.eth.BlockChain().CurrentBlock().Hash()
+}
+
+// Rollback un-sends previously added transactions.
+func (c *SimulatedBeacon) Rollback() {
+	// Flush all transactions from the transaction pools
+	maxUint256 := new(big.Int).Sub(new(big.Int).Lsh(common.Big1, 256), common.Big1)
+	c.eth.TxPool().SetGasTip(maxUint256)
+	// Set the gas tip back to accept new transactions
+	// TODO (Marius van der Wijden): set gas tip to parameter passed by config
+	c.eth.TxPool().SetGasTip(big.NewInt(params.GWei))
+}
+
+// Fork sets the head to the provided hash.
+func (c *SimulatedBeacon) Fork(parentHash common.Hash) error {
+	if len(c.eth.TxPool().Pending(false)) != 0 {
+		return errors.New("pending block dirty")
+	}
+	parent := c.eth.BlockChain().GetBlockByHash(parentHash)
+	if parent == nil {
+		return errors.New("parent not found")
+	}
+	return c.eth.BlockChain().SetHead(parent.NumberU64())
+}
+
+// AdjustTime creates a new block with an adjusted timestamp.
+func (c *SimulatedBeacon) AdjustTime(adjustment time.Duration) error {
+	if len(c.eth.TxPool().Pending(false)) != 0 {
+		return errors.New("could not adjust time on non-empty block")
+	}
+	parent := c.eth.BlockChain().CurrentBlock()
+	if parent == nil {
+		return errors.New("parent not found")
+	}
+	withdrawals := c.withdrawals.gatherPending(10)
+	return c.sealBlock(withdrawals, parent.Time+uint64(adjustment))
+}
```

Comment:
I think this function `AdjustTime` should either be renamed to `SealBlockWithTime(..)` or it should update parameter in the block generation (kind of like how the `chain_maker` does) so that on the next `SealBlock` it will seal with the provided timestamp. As written this name does not communicate to me clearly that a new block is about to be added to the chain.

Reply:
Again, this is an issue of backwards compatibility with the old sim backend